### PR TITLE
sql: remove entirely wrong comment

### PR DESCRIPTION
**Pull request by benesch**
We do exactly what this comment says we don't do: namely, hoisting all
CHECK constraint definitions to the table level and entirely ignoring
which columns are referenced.

Release note: None

### CODE REVIEW

File: pkg/sql/create_table.go [hunk 1]
```diff
@@ -1307,14 +1307,6 @@ func makeCheckConstraint(
 	semaCtx *tree.SemaContext,
 	evalCtx *tree.EvalContext,
 ) (*sqlbase.TableDescriptor_CheckConstraint, error) {
-	// CHECK expressions seem to vary across databases. Wikipedia's entry on
-	// Check_constraint (https://en.wikipedia.org/wiki/Check_constraint) says
-	// that if the constraint refers to a single column only, it is possible to
-	// specify the constraint as part of the column definition. Postgres allows
-	// specifying them anywhere about any columns, but it moves all constraints to
-	// the table level (i.e., columns never have a check constraint themselves). We
```
Comment (jordanlewis) [inline pkg/sql/create_table.go]: Hmm, I kind of like the rest of this comment. Would it be totally crazy to just modify the end to explain what we actually do?

Comment (dt) [reply pkg/sql/create_table.go]: Yeah, could just change the last sentence to "We do the same."

**Comment by benesch** [comment]
Ok, pushed a new rev that installs a rewording of the comment on `HoistConstraints`. I think that's a more logical place for the explanation.

**Review by dt — APPROVED** [review]
:plause:

**Comment by benesch** [comment]
TFTRs!

**Pull request merged**